File: Códigos/C_Codes/Revisão estruturas/revisaoEstruturas.c
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>
#include <string.h>
#include <math.h>
#include <time.h>
#include <locale.h>

#define ACELERATION_G 9.81
#define PI 3.141592653589793 // Definição de PI usando #define

// Definição do enum cores
enum cores {
    RED = 0,
    GREEN = 1,
    BLUE = 2
};

// Função para verificar se a aceleração é alta
bool is_high_acceleration(double acceleration) {
    return acceleration > ACELERATION_G;
}

// Função para retornar a cor associada com base na aceleração
enum cores get_color_based_on_acceleration(double acceleration) {
    if (acceleration < ACELERATION_G) {
        return GREEN; // Aceleração baixa é associada à cor GREEN
    } else if (acceleration == ACELERATION_G) {
        return BLUE;  // Aceleração igual à constante é associada à cor BLUE
    } else {
        return RED;   // Aceleração alta é associada à cor RED
    }
}

// Declaração da variável do tipo enum cores
enum cores colours;

// Declaração da variável do tipo union
union {
    int numero;
    float decimal;
    char letra;
} u;

// Definição da struct pessoa
struct pessoa {
    int idade;
    char nome[30];
    float estatura;
};

// Declaração da variável do tipo struct pessoa
struct pessoa Pessoa;

// Definição de um typedef para Livro
typedef struct {
    char titulo[100];
    char autor[50];
    int ano_publicacao;
    float preco;
} Livro;

// Declaração de uma variável do tipo Livro
Livro Livro1;

// Declaração de variáveis de tipos diversos
unsigned short int serie;
unsigned long int max32 = 4294967295UL;
const float PI_F = 3.1415927f;
char projeto[] = "Ampliar Tecnologias";

int main(void) {
    setlocale(LC_ALL, "Portuguese");
    srand((unsigned int)time(NULL));

    colours = RED;
    printf("Cor selecionada: %d\n", colours);

    u.numero = 42;
    printf("Valor da union (numero): %d\n", u.numero);

    u.decimal = 3.14f;
    printf("Valor da union (decimal): %.2f\n", u.decimal);

    u.letra = 'A';
    printf("Valor da union (letra): %c\n", u.letra);

    Pessoa.idade = 25;
    strcpy(Pessoa.nome, "João Silva");
    Pessoa.estatura = 1.75f;
    printf("Pessoa:\n");
    printf("Idade: %d\n", Pessoa.idade);
    printf("Nome: %s\n", Pessoa.nome);
    printf("Estatura: %.2f\n", Pessoa.estatura);

    strcpy(Livro1.titulo, "O Guia do Mochileiro das Galáxias");
    strcpy(Livro1.autor, "Douglas Adams");
    Livro1.ano_publicacao = 1979;
    Livro1.preco = 29.90f;
    printf("Livro:\n");
    printf("Título: %s\n", Livro1.titulo);
    printf("Autor: %s\n", Livro1.autor);
    printf("Ano de publicação: %d\n", Livro1.ano_publicacao);
    printf("Preço: %.2f\n", Livro1.preco);

    serie = 12345;
    printf("Número da série: %hu\n", serie);
    printf("Valor máximo de 32 bits: %lu\n", max32);
    printf("Valor de PI_F: %.7f\n", PI_F);
    printf("Valor de PI: %.15f\n", PI);
    printf("Projeto: %s\n", projeto);

    double acceleration = 12.5;

    if (is_high_acceleration(acceleration)) {
        printf("A aceleração de %.2f é alta.\n", acceleration);
    } else {
        printf("A aceleração de %.2f não é alta.\n", acceleration);
    }

    enum cores color = get_color_based_on_acceleration(acceleration);

    switch (color) {
        case RED:
            printf("A cor associada à aceleração é RED.\n");
            break;
        case GREEN:
            printf("A cor associada à aceleração é GREEN.\n");
            break;
        case BLUE:
            printf("A cor associada à aceleração é BLUE.\n");
            break;
        default:
            printf("Cor desconhecida.\n");
            break;
    }

    printf("Números aleatórios:\n");
    for (int i = 0; i < 5; ++i) {
        int random_number = rand() % 100; // Número aleatório entre 0 e 99
        printf("%d ", random_number);
    }
    printf("\n");

    return 0;
}

File: Códigos/Python _codes/Revisão Estruturas/revisao_estruturas.py
import random
import sys
import locale

sys.stdout.reconfigure(encoding='utf-8')

# Definições constantes
ACELERATION_G = 9.81
PI = 3.141592653589793

# Definição de enum com um dicionário
CORES = {
    'RED': 0,
    'GREEN': 1,
    'BLUE': 2
}

# Função para verificar se a aceleração é alta
def is_high_acceleration(acceleration):
    return acceleration > ACELERATION_G

# Função para retornar a cor associada com base na aceleração
def get_color_based_on_acceleration(acceleration):
    if acceleration < ACELERATION_G:
        return 'GREEN'  # Aceleração baixa associada à cor GREEN
    elif acceleration == ACELERATION_G:
        return 'BLUE'   # Aceleração igual à constante associada à cor BLUE
    else:
        return 'RED'    # Aceleração alta associada à cor RED

# Definição da union usando uma classe
class Union:
    def __init__(self):
        self.numero = None
        self.decimal = None
        self.letra = None

# Definição da struct pessoa usando uma classe
class Pessoa:
    def __init__(self, idade, nome, estatura):
        self.idade = idade
        self.nome = nome
        self.estatura = estatura

# Definição da struct Livro usando uma classe
class Livro:
    def __init__(self, titulo, autor, ano_publicacao, preco):
        self.titulo = titulo
        self.autor = autor
        self.ano_publicacao = ano_publicacao
        self.preco = preco

# Definição de variáveis
serie = 12345
max32 = 4294967295
PI_F = 3.1415927
projeto = "Ampliar Tecnologias"

# Configurações de localidade (equivalente ao setlocale)
locale.setlocale(locale.LC_ALL, 'pt_BR.UTF-8')

# Uso das variáveis e classes
colours = CORES['RED']
print(f"Cor selecionada: {colours}")

u = Union()
u.numero = 42
print(f"Valor da union (numero): {u.numero}")

u.decimal = 3.14
print(f"Valor da union (decimal): {u.decimal:.2f}")

u.letra = 'A'
print(f"Valor da union (letra): {u.letra}")

# Instância da struct pessoa
pessoa = Pessoa(25, "João Silva", 1.75)
print(f"Pessoa:\nIdade: {pessoa.idade}\nNome: {pessoa.nome}\nEstatura: {pessoa.estatura:.2f}")

# Instância da struct Livro
livro1 = Livro("O Guia do Mochileiro das Galáxias", "Douglas Adams", 1979, 29.90)
print(f"Livro:\nTítulo: {livro1.titulo}\nAutor: {livro1.autor}\nAno de publicação: {livro1.ano_publicacao}\nPreço: {livro1.preco:.2f}")

# Exibição de outras variáveis
print(f"Número da série: {serie}")
print(f"Valor máximo de 32 bits: {max32}")
print(f"Valor de PI_F: {PI_F:.7f}")
print(f"Valor de PI: {PI:.15f}")
print(f"Projeto: {projeto}")

# Verificando aceleração
acceleration = 12.5
if is_high_acceleration(acceleration):
    print(f"A aceleração de {acceleration:.2f} é alta.")
else:
    print(f"A aceleração de {acceleration:.2f} não é alta.")

# Cor associada à aceleração
color = get_color_based_on_acceleration(acceleration)

if color == 'RED':
    print("A cor associada à aceleração é RED.")
elif color == 'GREEN':
    print("A cor associada à aceleração é GREEN.")
elif color == 'BLUE':
    print("A cor associada à aceleração é BLUE.")
else:
    print("Cor desconhecida.")

# Gerando números aleatórios
print("Números aleatórios:")
for _ in range(5):
    random_number = random.randint(0, 99)  # Número aleatório entre 0 e 99
    print(random_number, end=' ')
print()